src/server/api/controllers/vessel-reports.controller.js

```diff
@@ -0,0 +1,77 @@
+"use strict";
+
+const HttpError = require("../lib/utils/http-error");
+const knex = require("../../config/db");
+
+//Get vessels reports by vessel id
+const getVesselsReportByVesselId = async id => {
+  const vesselReports = await knex
+    .from("vessel_reports")
+    .where({ vessel_id: id });
+  if (vesselReports.length === 0) {
+    return `no vessel exist with that id ${id} `;
+  }
+  return vesselReports;
+};
+
+//create vessel-reports
+const createVesselReport = async ({ body }) => {
+  const {
+    vessel_id,
+    position_waypoints,
+    current_speed,
+    hfo_consumption,
+    lsfo_consumption
+  } = body;
+
+  const vessel = await knex
+    .from("vessels")
+    .select("*")
+    .where({
+      id: vessel_id
+    });
```

Comment:
You have a problem with this query. I'm not quite sure why:
```
Error: Undefined binding(s) detected when compiling SELECT. Undefined column(s): [id] query: select `vessels`.`id` as `id` from `vessels` where `id` = ?
```

Reply:
I spend some more time debugging this and it turns out that `body` is undefined. It's too late for me to debug further, but have a look at the router as well where it comes from.

Reply:
I am not sure why you are not getting. But I am getting all input. Maybe be in the body of postman , you are giving "id" also. I think we should not give id in body because that has to create by itself. 

-